src/indexers/governance-proposals-calldata-indexer.ts

```diff
@@ -0,0 +1,113 @@
+// optimize getContractAbi, so we don't fetch the same contract abi twice
+
+import { NETWORK_CONFIGS } from '@badger-dao/sdk';
+import { GovernanceProposalsDecodedData } from '@badger-dao/sdk/lib/api/interfaces/governance-proposals-decoded-data.interface';
+import { Network } from '@badger-dao/sdk/lib/config/enums/network.enum';
+import { Provider } from '@ethersproject/providers';
+import { ethers } from 'ethers';
+import detectProxyTarget from 'ethers-proxies';
+
+import { getDataMapper } from '../aws/dynamodb.utils';
+import { getOrCreateChain } from '../chains/chains.utils';
+import { getProposalsWithEmptyDecodedCallData } from '../governance/governance.utils';
+import { extendAbiMethodsWithMeta } from '../utils/contract.utils';
+import { rfw } from '../utils/retry.utils';
+import { formScanApiUrl, getContractAbi } from '../utils/scans.utils';
+
+export async function decodeGavernanceProposalsCallData() {
+  const proposals = await getProposalsWithEmptyDecodedCallData();
+
+  if (proposals.length === 0) {
+    console.log('No proposals with empty decoded call data found');
+    return;
+  }
+
+  const mapper = getDataMapper();
+
+  for (const proposal of proposals) {
+    const chain = getOrCreateChain(<Network>proposal.network);
+
+    const scanApiUrl = formScanApiUrl(NETWORK_CONFIGS[chain.network].explorerUrl);
+
+    const decodedTargetMethod = await getTargetMethodDecoded(
```

Comment:
if this is being used as a hash key then proposals calling the same function with the same paratmers will result in a non unique result for that index query

maybe you can consider making the hash key a composite key utilizing `bloch.hash`, or `transaction.hash` along with the log index to form unique hash keys.

Reply:
i dont think we will query by callData, more like by proposalId and network, i've created this index with only purpose to find non decoded entities

do u think we should do this key uniq?